app/models/grda_warehouse/tasks/client_cleanup.rb

```diff
@@ -464,22 +464,48 @@ def choose_best_pronouns dest_attr, source_clients
       dest_attr
     end
 
+    # Get the best SSN (has value and quality is full or partial, oldest breaks the tie)
+    # note, source clients might be just an array of hashes
+    SSN_DQS = [1, 2, 8, 9, 99].to_set.freeze
     def choose_best_ssn dest_attr, source_clients
-      # Get the best SSN (has value and quality is full or partial, oldest breaks the tie)
-      non_blank_ssn = source_clients.select { |sc| sc[:SSN].present? }
-      if non_blank_ssn.any?
-        best = nil
-        [1, 2, 8, 9, 99].each do |dq_value|
-          best = non_blank_ssn.select { |sc| sc[:SSNDataQuality] == dq_value }.min_by { |sc| sc[:DateCreated] }
-          break if best.present?
-        end
-        best = non_blank_ssn.min_by { |sc| sc[:DateCreated] } if best.blank?
+      items = source_clients.
+        filter { |sc| sc[:SSNDataQuality].in?(SSN_DQS) }.
```

Comment:
Not sure we want to throw out any source clients where we are missing SSNDataQuality.  Historically, these data quality fields have been all over the map.  Can we ignore blank `SSN` and set `SSNDataQuality` to partial if `SSN` is present, but `SSNDataQuality` is blank?